nsxt/segment_common.go

```diff
@@ -779,14 +814,29 @@ func nsxtPolicySegmentDelete(d *schema.ResourceData, m interface{}) error {
 		Pending: pendingStates,
 		Target:  targetStates,
 		Refresh: func() (interface{}, string, error) {
-			ports, err := portsClient.List(id, nil, nil, nil, nil, nil, nil)
-			if err != nil {
-				return ports, "error", logAPIError("Error listing segment ports", err)
+			var ports interface{}
+			var numOfPorts int
+			if isPolicyGlobalManager(m) {
+				portsClient := gm_segments.NewDefaultPortsClient(connector)
```

Comment:
I wonder if the port gets propagated to global manager?

Reply:
I'm not sure what you mean. You can create ports on the GM, so you should be able to wait for their deletion.

Reply:
This code was added to deal with automatic port creation for vsphere vms. I'm not sure global manager will see these port. But this is something we can test later.